samples/cpp/speech_sample/main.cpp

```diff
@@ -243,9 +243,12 @@ int main(int argc, char* argv[]) {
         }
         gnaPluginConfig[ov::hint::inference_precision.name()] = (FLAGS_qb == 8) ? ov::element::i8 : ov::element::i16;
         auto parse_target = [&](const std::string& target) -> ov::intel_gna::HWGeneration {
-            return (target == "GNA_TARGET_2_0") ? ov::intel_gna::HWGeneration::GNA_2_0
-                                                : (target == "GNA_TARGET_3_0") ? ov::intel_gna::HWGeneration::GNA_3_0
-                                                                               : ov::intel_gna::HWGeneration::UNDEFINED;
+            return (target == "GNA_TARGET_2_0")
+                       ? ov::intel_gna::HWGeneration::GNA_2_0
+                       : (target == "GNA_TARGET_3_0")
+                             ? ov::intel_gna::HWGeneration::GNA_3_0
+                             : (target == "GNA_TARGET_3_5") ? ov::intel_gna::HWGeneration::GNA_3_5
+                                                            : ov::intel_gna::HWGeneration::UNDEFINED;
         };
```

Comment:
Please extract this into separate function and add log from speech sample informing which ov::intel_gna::HWGeneration was chosen.
(Currently, if wrong string e.g., GNA_TARGET_X_Y is given, it will be silently replaced with UNDEFINED which may by misleading as one may think that X.Y is in fact supported).